vnext/Microsoft.ReactNative/ReactHost/ReactInstanceWin.cpp

```diff
@@ -482,9 +484,16 @@ facebook::react::NativeLoggingHook ReactInstanceWin::GetLoggingCallback() noexce
 std::function<void(facebook::react::JSExceptionInfo &&)> ReactInstanceWin::GetJSExceptionCallback() noexcept {
   if (m_options.OnJSException) {
     return CreateExceptionCallback(Mso::Copy(m_options.OnJSException));
+  } else if (m_options.DeveloperSettings.IsDevModeEnabled) {
+    auto localWkReactHost = m_weakReactHost;
+    return CreateRedBoxExceptionCallback(m_uiMessageThread.LoadWithLock(), [localWkReactHost]() {
```

Comment:
>localWkReactHost [](start = 76, length = 16)

Consider to assign it here in-place or std::move it here to avoid extra ref-count add/release for the weak reference.